cluster-operator/src/main/java/io/strimzi/operator/cluster/model/Certificates.java

```diff
@@ -0,0 +1,274 @@
+        this(certManager, clusterName, secrets, false);
+    }
+
+    public Certificates(CertManager certManager, String clusterName, List<Secret> secrets, boolean needsRenewal) {
+        this.certManager = certManager;
+        this.clusterCaCertRenewed = needsRenewal;
+        Secret clusterCa = null;
+        Secret clientsCa = null;
+        Secret brokersSecret = null;
+        Secret eoSecrets = null;
+        Secret toSecret = null;
+        Secret zkNodesSecret = null;
+        for (Secret secret: secrets) {
+            String name = secret.getMetadata().getName();
+            if (AbstractModel.getClusterCaName(clusterName).equals(name)) {
+                clusterCa = secret;
+            } else if (KafkaCluster.clientsCASecretName(clusterName).equals(name)) {
+                clientsCa = secret;
+            } else if (KafkaCluster.brokersSecretName(clusterName).equals(name)) {
+                brokersSecret = secret;
+            } else if (EntityOperator.secretName(clusterName).equals(name)) {
+                eoSecrets = secret;
+            } else if (TopicOperator.secretName(clusterName).equals(name)) {
+                toSecret = secret;
+            } else if (ZookeeperCluster.nodesSecretName(clusterName).equals(name)) {
+                zkNodesSecret = secret;
+            }
+        }
+        if (clusterCa == null) {
+            throw new NoCertificateSecretException("The cluster CA certificate Secret is missing");
+        } else if (clusterCa.getData().get("cluster-ca.key") == null) {
+            throw new NoCertificateSecretException("The cluster CA certificate Secret is missing data.cluster-ca.key");
+        } else if (clusterCa.getData().get("cluster-ca.crt") == null) {
+            throw new NoCertificateSecretException("The cluster CA certificate Secret is missing data.cluster-ca.crt");
+        }
+        this.clusterCa = asCertAndKey(clusterCa, "cluster-ca.key", "cluster-ca.crt");
+        this.clientsCa = asCertAndKey(clientsCa, "clients-ca.key", "clients-ca.crt");
+        this.brokersSecret = brokersSecret;
+        this.eoCertAndKey = asCertAndKey(eoSecrets, "entity-operator.key", "entity-operator.crt");
+        this.toCertAndKey = asCertAndKey(toSecret, "entity-operator.key", "entity-operator.crt");
+        this.zkNodesSecret = zkNodesSecret;
+    }
+
+    private CertAndKey asCertAndKey(Secret secret, String key, String cert) {
+        Base64.Decoder decoder = Base64.getDecoder();
+        return secret == null ? null : new CertAndKey(
+                decoder.decode(secret.getData().get(key)),
+                decoder.decode(secret.getData().get(cert)));
+    }
+
+    public CertAndKey clusterCa() {
+        return clusterCa;
+    }
+
+    public CertAndKey clientsCa(Kafka kafka) throws IOException {
+        if (clientsCa == null) {
+            clientsCa = generateClientsCa(kafka);
+        }
+        return clientsCa;
+    }
+
+    public CertAndKey toCert(Kafka kafka) throws IOException {
```

Comment:
I would use a clearer name as `topicOperatorCert`. It is not so clear that `to` refers to the TO and not the English word "to"